src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -867,84 +881,51 @@ private async Task CommitCoreAsync(IUIThreadOperationContext operationContext, b
             }
 
             // The user hasn't canceled by now, so we're done waiting for them. Off to rename!
-            using var _ = operationContext.AddScope(allowCancellation: false, EditorFeaturesResources.Updating_files);
+            using var _1 = operationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Updating_files);
```

Comment:
changed this logic around slightly (to also allow more time for user to cancel if they want).  Stages are:

1. compute tree/text changes to make to the workspace. (still cancellable)
2. jump to ui thread.
3. make things non-cancellable.
4. pre-cleanup (rollback to before linked edits were made)
5. try apply to workspace.
6. if that fails, try again, in case our rename notification caused a change with a 3rd party.
7. postcleanup.